data-connection-jdbc/src/main/java/io/micronaut/data/connection/jdbc/operations/DefaultDataSourceConnectionOperations.java

```diff
@@ -87,4 +115,19 @@ protected void closeConnection(ConnectionStatus<Connection> connectionStatus) {
         }
     }
 
+    /**
+     * Checks whether current connection is Oracle database connection.
+     *
+     * @param connection The connection
+     * @return true if current connection is Oracle database connection
+     */
+    private boolean isOracleConnection(Connection connection) {
+        try {
+            String databaseProductName = connection.getMetaData().getDatabaseProductName();
+            return StringUtils.isNotEmpty(databaseProductName) && databaseProductName.toUpperCase().contains("ORACLE");
```

Comment:
Not sure if this is reliable enough, but should be and we have no info which db connection is used here.

Reply:
use `equalsIgnoreCase` instead and cache the result somewhere